tough/src/lib.rs

```diff
@@ -94,45 +95,127 @@ impl From<ExpirationEnforcement> for bool {
-    /// becomes out of date later; the client establishes trust up to the most recent root.json
-    /// file.)
-    pub root: R,
+pub struct RepositoryLoader<R>
+where
+    R: Read,
+{
+    root: R,
+    metadata_base_url: Url,
+    targets_base_url: Url,
+    transport: Option<Box<dyn Transport>>,
+    limits: Option<Limits>,
+    datastore: Option<PathBuf>,
+    expiration_enforcement: Option<ExpirationEnforcement>,
+}
 
-    /// Tough stores the most recently fetched timestamp, snapshot, and targets metadata files here
-    /// to detect version rollback attacks.
+impl<R: Read> RepositoryLoader<R> {
+    /// Create a new `RepositoryLoader`.
     ///
-    /// You may chose to provide a [`PathBuf`] to a directory on a persistent filesystem where
-    /// directory must exist prior to calling [`Repository::load`]. If given `None`, a temporary
-    /// directory will be created and cleaned up for for you.
-    pub datastore: Option<PathBuf>,
+    /// `root` is a [`Read`]er for the trusted root metadata file, which you must ship with your
+    /// software using an out-of-band process. It should be a copy of the most recent root.json
+    /// from your repository. (It's okay if it becomes out of date later; the client establishes
+    /// trust up to the most recent root.json file.)
+    ///
+    /// `metadata_base_url` and `targets_base_url` are the base URLs where the client can find
+    /// metadata (such as root.json) and targets (as listed in targets.json).
+    pub fn new(root: R, metadata_base_url: Url, targets_base_url: Url) -> Self {
+        Self {
+            root,
+            metadata_base_url,
+            targets_base_url,
+            transport: None,
+            limits: None,
+            datastore: None,
+            expiration_enforcement: None,
+        }
+    }
 
-    /// The URL base for TUF metadata (such as timestamp.json).
-    pub metadata_base_url: String,
+    /// Load and verify TUF repository metadata.
+    pub fn load(self) -> Result<Repository> {
+        Repository::load(self)
+    }
 
-    /// The URL base for targets.
-    pub targets_base_url: String,
+    /// Set the transport. If no transport has been set, [`DefaultTransport`] will be used.
+    pub fn transport<T: Transport + 'static>(mut self, transport: T) -> Self {
+        self.transport = Some(Box::new(transport));
+        self
+    }
 
-    /// Limits used when fetching repository metadata.
-    ///
-    /// This parameter implements [`Default`]; see its documentation for details.
-    pub limits: Limits,
+    /// Set a the repository [`Limits`].
+    pub fn limits(mut self, limits: Limits) -> Self {
+        self.limits = Some(limits);
+        self
+    }
 
-    /// Metadata expiration enforcement.
-    ///
-    /// When this parameter is set to `Unsafe`, a `Repository` will succeed in loading even if it
-    /// has expired metadata files.
+    /// Optionally set a `datastore` directory path. `datastore` is a directory on a persistent
```

Comment:
```suggestion
    /// Set a `datastore` directory path. `datastore` is a directory on a persistent
```
If you call this, you're setting it :)

Reply:
Fixed in [6ce9dc8](https://github.com/awslabs/tough/compare/231ceaa8956d1fadd85c58e7f8202daff603bd5b..6ce9dc89c1658d2a3a11c9b6ef847cfe10f8a3fd)